bitcoin/src/crypto/key.rs

```diff
@@ -332,13 +366,40 @@ impl fmt::Display for CompressedPublicKey {
 }
 
 impl FromStr for CompressedPublicKey {
-    type Err = Error;
+    type Err = CompressedPublicKeyParseError;
```

Comment:
Should probably be `ParseCompressedPublicKeyError` so as to be uniform with `ParsePublicKeyError`.